docs/proposed/psa-driver-interface.md

```diff
@@ -321,6 +321,148 @@ TODO: key input and output for opaque drivers; deterministic key generation for
 
 TODO
 
+### Driver entry points for PAKE
+
+PAKE operation is divided into two stages: collecting inputs and computation. Core side is responsible for keeping inputs and core set-data functions do not have driver entry points. Collected inputs are available for drivers via get-data functions for `password`, `role` and `cipher_suite`.
+
+### PAKE driver dispatch logic
+The core decides whether to dispatch a PAKE operation to a driver based on the location of the provided password.
+When all inputs are collected and `"psa_pake_output"` or `"psa_pake_input"` is called for the first time `"pake_setup"` driver entry point is invoked.
+
+1. Lifetime of the `password` is local storage
+- if there is a transparent driver available for the given configuration, the core calls that driver's `"pake_setup"` and subsequent entry points.
+- if a transparent driver is not available or can not handle a given configuration, the core uses its built-in implementation.
+2. Lifetime of the `password` is test driver
```

Comment:
Should this be 'opaque driver' instead of 'test driver'?

Reply:
Yes, it seems `opaque driver` is more suitable, but looking at the code we have:
```
        case PSA_CRYPTO_TEST_DRIVER_LOCATION:  <---- 
            status = mbedtls_test_opaque_pake_setup(
                        &operation->data.ctx.opaque_test_driver_ctx,
                        inputs );
             if( status == PSA_SUCCESS )
                operation->id = MBEDTLS_TEST_OPAQUE_DRIVER_ID;
            return status;
```